### PR TITLE
basic models, relations and queries

### PR DESCRIPTION
**Pull request by ankitj**
added queries

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -62,4 +62,12 @@ dependencies {
...
     compile 'com.android.support.constraint:constraint-layout:1.0.2'
     testCompile 'junit:junit:4.12'
+    // parse
+    compile 'com.parse.bolts:bolts-android:1.4.0'
+    compile 'com.parse:parse-android:1.14.1'
+    compile 'com.parse:parseinterceptors:0.0.2'
+    compile 'com.parse:parsefacebookutils-v4-android:1.10.4@aar'
+    // butterknife
+    compile 'com.jakewharton:butterknife:8.4.0'
+    annotationProcessor 'com.jakewharton:butterknife-compiler:8.4.0'
```
Comment (gbajaj) [inline app/build.gradle:72]: @ankitj lets not use better knife and start using DataBinding as that's more official way for doing the same thing

File: app/src/main/java/com/codepath/hungrybird/HungryBirdApplication.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        // clientKey is not needed unless explicitly configured
+        // any network interceptors must be added with the Configuration Builder given this syntax
+        String PARSE_SERVER_URL = "https://parse-demo-2.herokuapp.com/parse/";
+        String PARSE_APP_ID = "parse-demo-2-app-id";
+        Parse.initialize(new Parse.Configuration.Builder(this)
+                .applicationId(PARSE_APP_ID) // should correspond to APP_ID env variable
+                .clientKey(null)  // set explicitly unless clientKey is explicitly configured on Parse server
+                .addNetworkInterceptor(new ParseLogInterceptor())
+                .server(PARSE_SERVER_URL).build());
+
```
Comment (gbajaj) [inline app/src/main/java/com/codepath/hungrybird/HungryBirdApplication.java:38]: How are we deploying the parse server?

File: app/src/main/java/com/codepath/hungrybird/model/Dish.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    public String getCuisine() {
+        return getString("cuisine");
+    }
+
+    public static enum Cuisine {
+        INDIAN("indian"),
+        ITALIAN("italian");
+
```
Comment (gbajaj) [inline app/src/main/java/com/codepath/hungrybird/model/Dish.java:40]: Every time we have add a cuisine  we would have to change code here, there should be a better way than this to add more cuisines

File: app/src/main/java/com/codepath/hungrybird/HungryBirdApplication.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        super.onCreate();
+
+        // Use for troubleshooting -- remove this line for production
+        Parse.setLogLevel(Parse.LOG_LEVEL_DEBUG);
+
+        registerModels();
+
+        // set applicationId, and server server based on the values in the Heroku settings.
+        // clientKey is not needed unless explicitly configured
+        // any network interceptors must be added with the Configuration Builder given this syntax
```
Comment (gbajaj) [inline app/src/main/java/com/codepath/hungrybird/HungryBirdApplication.java:30]: HungryBirdApplication is already in develop branch, please use that

File: app/src/main/java/com/codepath/hungrybird/model/Dish.java (hunk 3 above)
Comment (dhwanishah) [reply app/src/main/java/com/codepath/hungrybird/model/Dish.java:40]: Also, using enums is a bad idea for efficiency sake per android. If we don't have that much abstraction, then just use int vars. Better for memory: https://developer.android.com/topic/performance/memory.html#Overhead

**Pull request merged**